engine/src/resource/resource.cpp

```diff
@@ -143,19 +143,19 @@ void Resource<T>::Zero() {
 // TODO: implement this so it would actually work
 // Currently it isn't called. Possibly because a copy constructor is called instead or something
 
-/*template<typename T>
-Resource<T> Resource<T>::operator=(const T &value) {
-    std::cout << value << "\n";
-    value_ = value;
-    std::cout << value_ << "\n";
-    if(max_value_ != -1) {
-        value_ = std::min(max_value_, value_);
-    }
-    std::cout << value_ << "\n";
-    value_ = std::max(min_value_, value_);
-    std::cout << value_ << "\n";
-    return *this;
-}*/
+//template<typename T>
```

Comment:
Looks like we either should take this out or it should be `#ifdef/#endif` enabled, but most likely it needs to just go away